python/pyarrow/dataset.py

```diff
@@ -486,8 +488,13 @@ def parquet_dataset(metadata_path, schema=None, filesystem=None, format=None):
         filesystem, _ = _ensure_filesystem(filesystem)
 
     metadata_path = _normalize_path(filesystem, _stringify_path(metadata_path))
+    options = ParquetFactoryOptions(
+        partition_base_dir=partition_base_dir,
+        partitioning=partitioning
```

Comment:
Can you add here a `partitioning = _ensure_partitioning(partitioning)` before passing it to ParquetFactoryOptions? 

(that will eg make sure that `parquet_dataset(..., partitioning="hive")` will also work, similarly as for `dataset(..)`)